web/server/routers/post.ts

```diff
@@ -198,6 +199,22 @@ export const postRouter = router({
       }
       return { success: true };
     }),
+  delete: procedure
+    .input(
+      z.object({
+        postOid: zodOidType,
+      })
+    )
+    .mutation(async ({ input }) => {
+      const deleteSuccess = await deletePost(input.postOid);
+      if (!deleteSuccess.success) {
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: "An unexpected error occurred.",
+        });
+      }
+      return { success: true };
```

Comment:
Wrap in a try catch instead of checking the `success` value. Add logic to throw the error directly if it is an instance of `TRPCError`, and internal server error otherwise